app/javascript/mastodon/features/ui/components/image_loader.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import ImmutablePureComponent from 'react-immutable-pure-component';
+import PropTypes from 'prop-types';
+
+class ImageLoader extends ImmutablePureComponent {
+
+  static propTypes = {
+    src: PropTypes.string.isRequired,
+  }
+
+  state = {
+    loading: true,
+    error: false,
+  }
+
+  constructor(props) {
+    super(props);
+    this.loadImage(props.src);
+  }
```

Comment:
In that case you should use componentWillMount. setState in constructor is not very good.